src/Test.Utilities/CodeFixTestBase.cs

```diff
@@ -146,6 +167,37 @@ private void VerifyFix(string language, DiagnosticAnalyzer analyzerOpt, CodeFixP
             Assert.Equal(newSource, actualText.ToString());
         }
 
+        private void VerifyFixAll(
+            Document document,
+            DiagnosticAnalyzer analyzerOpt,
+            CodeFixProvider codeFixProvider,
+            string newSource,
+            string newSourceFileName,
+            IEnumerable<TestAdditionalDocument> additionalFiles,
+            int? codeFixIndex,
+            bool allowNewCompilerDiagnostics,
+            bool onlyFixFirstFixableDiagnostic,
+            TestValidationMode validationMode)
+        {
+            var fixableDiagnosticIds = codeFixProvider.FixableDiagnosticIds.ToSet();
+            Func<IEnumerable<Diagnostic>, ImmutableArray<Diagnostic>> getFixableDiagnostics = diags =>
+                diags.Where(d => fixableDiagnosticIds.Contains(d.Id)).ToImmutableArrayOrEmpty();
+
+            var analyzerDiagnostics = GetSortedDiagnostics(analyzerOpt, new[] { document }, additionalFiles: additionalFiles, validationMode: validationMode);
+            var compilerDiagnostics = document.GetSemanticModelAsync().Result.GetDiagnostics();
+            var fixableDiagnostics = getFixableDiagnostics(analyzerDiagnostics.Concat(compilerDiagnostics));
+
+            var fixAllProvider = codeFixProvider.GetFixAllProvider();
+            var diagnosticProvider = new FixAllDiagnosticProvider(analyzerOpt, additionalFiles, validationMode, getFixableDiagnostics);
+            var fixAllContent = new FixAllContext(document, codeFixProvider, FixAllScope.Document, string.Empty, fixableDiagnostics.Select(d => d.Id), diagnosticProvider, CancellationToken.None);
```

Comment:
Can you please open a test infrastructure issue to track adding support FixAll tests for different scopes (project, solution)?

Reply:
nit: `fixAllContext`

Reply:
[https://github.com/dotnet/roslyn-analyzers/issues/1391](https://github.com/dotnet/roslyn-analyzers/issues/1391)

---
In reply to: [152385767](https://github.com/dotnet/roslyn-analyzers/pull/1336#discussion_r152385767) [](ancestors = 152385767)